src/main/java/net/minecraftforge/fluids/capability/templates/FluidHandlerConcatenate.java

```diff
@@ -0,0 +1,109 @@
+package net.minecraftforge.fluids.capability.templates;
+
+import com.google.common.collect.Lists;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.FluidTankInfo;
+import net.minecraftforge.fluids.capability.IFluidHandler;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * FluidHandlerConcatenate is a template class for concatenating multiple handlers into one.
+ * If each tank is restricted to exactly one type of fluid, then use {@link FluidHandlerFluidMap} as it is more efficient.
+ */
+public class FluidHandlerConcatenate implements IFluidHandler
+{
+    protected final IFluidHandler[] subHandlers;
+
+    public FluidHandlerConcatenate(IFluidHandler... subHandlers)
+    {
+        this.subHandlers = subHandlers;
+    }
+
+    public FluidHandlerConcatenate(Collection<IFluidHandler> subHandlers)
+    {
+        this.subHandlers = subHandlers.toArray(new IFluidHandler[subHandlers.size()]);
+    }
+
+    @Override
+    public FluidTankInfo[] getTankInfo()
+    {
+        List<FluidTankInfo> tanks = Lists.newArrayList();
+        for (IFluidHandler handler : subHandlers)
+        {
+            Collections.addAll(tanks, handler.getTankInfo());
+        }
+        return tanks.toArray(new FluidTankInfo[tanks.size()]);
+    }
+
+    @Override
+    public int fill(FluidStack resource, boolean doFill)
+    {
+        int k = 0;
+        for (IFluidHandler handler : subHandlers)
+        {
+            k += handler.fill(resource, doFill);
```

Comment:
Won't this break horribly and cause dupe bugs as k is never subtracted from the resource?